cmd/autoconfigbrancher/main.go

```diff
@@ -265,3 +259,39 @@ func main() {
 		logrus.WithError(err).Fatal("PR creation failed.")
 	}
 }
+
+func runSteps(steps []step, author string, stdout, stderr io.Writer) (needsPushing bool, err error) {
+	startCommitOut, err := exec.Command("git", "rev-parse", "HEAD").CombinedOutput()
+	if err != nil {
+		return false, fmt.Errorf("failed to execute `git rev-parse HEAD`: %w\noutput:%s\n", err, string(startCommitOut))
+	}
+	startCommitSHA := strings.TrimSpace(string(startCommitOut))
+
+	var commitsCounter int
+	for _, step := range steps {
+		committed, err := runAndCommitIfNeeded(stdout, stderr, author, step.command, step.arguments)
+		if err != nil {
+			return false, fmt.Errorf("failed to run command and commit the changes: %w", err)
+		}
+
+		if committed {
+			commitsCounter++
+		}
+	}
+
+	if commitsCounter == 0 {
+		logrus.Info("No new commits")
+		return false, nil
+	}
+
+	overallDiff, err := exec.Command("git", "diff", startCommitSHA).CombinedOutput()
```

Comment:
`git diff` has a `--quiet` option for this, or we could compare tree hashes directly (the Jira issue has examples of the latter).

Reply:
I don't see anything with `--quiet` in https://issues.redhat.com/browse/DPTP-2394 ?

The reason I chose to compare the output and not the exit code is that I want to be able to differentiate "Executing the command failed" from "There is a diff".

Reply:
"The latter" referred to comparing tree hashes (i.e. `--format=%T`).

There are distinct exit codes for errors:

```
$ git diff --quiet; echo $?
1
$ git diff --quiet 404 --; echo $?
fatal: bad revision '404'
128
$ git -C /404 diff --quiet; echo $?
fatal: cannot change to '/404': No such file or directory
128
```

```
$ git stash push --quiet
$ git diff --quiet; echo $?
0
$ git diff --quiet 404 --; echo $?
fatal: bad revision '404'
128
$ git -C 404 diff --quiet; echo $?
fatal: cannot change to '404': No such file or directory
128
```

Reply:
Yeah, but that makes for both more complicated code as I have to assert the returned error to get the error code and makes it harder to understand, because most ppl do not know about the different exit codes of `git diff` while OTOH it provides no advantage (we still have to have an output buffer in case the command fails, otherwise debugging is impossible).